aws-crt-http/HttpConnection.cs

```diff
@@ -1,8 +1,245 @@
-﻿using System;
+﻿/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+using System;
+using System.Collections;
+using System.Collections.Specialized;
+using System.Runtime.InteropServices;
+using System.Security;
+
+using Aws.Crt.IO;
 
 namespace Aws.Crt.Http
 {
-    public class HttpConnection
+    public delegate void OnConnectionSetup(int errorCode);
+    public delegate void OnConnectionShutdown(int errorCode);
+    public delegate void OnStreamOutgoingBody(HttpStream stream, out byte[] buffer);

```

Comment:
Let's chat about this. Maybe users should just set some kind of stream and we'll read it bit by bit for them. But we could still have an optional callback where they can do so window-adjustment stuff, if they desire.

we might move to an approach like that in C-land too. Brett is doing work on a streaming API for aws-c-io

Reply:
Discussed offline, we're going to mess with the API for body sending soon.